flake.nix

```
@@ -174,60 +158,19 @@
           pkgs.lib.listToAttrs
         ];
 
-        apps.tests = let
-          mkTest = {
-            name,
-            script,
-          }: let
-            test = pkgs.writeShellApplication {
-              inherit name;
-              text = ''
-                set -xeuo pipefail
-                WORKDIR=$(mktemp -d)
-                function clean() {
-                  rm -rf "''${WORKDIR}"
-                }
-                trap clean EXIT
-                cd "''${WORKDIR}"
-
-                ${script}
-              '';
-            };
-          in {
-            type = "app";
-            program = "${pkgs.lib.getExe test}";
-          };
-        in {
-          templates = pkgs.lib.flip pkgs.lib.mapAttrs self.templates (
-            name: _:
-              mkTest {
-                name = "test devshell ${name}";
-                script = ''
-                  nix flake new --template path:${./.}#${name} example --accept-flake-config
-
-                  pushd ./example
-                  # We test against the local version of `nickel-nix`, not the one in main (hence the --override-input).
-                  nix flake lock --override-input nickel-nix path:${./.} --accept-flake-config
-                  nix run .#regenerate-lockfile --accept-flake-config
-                  nix develop --accept-flake-config --print-build-logs < /dev/null
-                  popd
-                '';
-              }
-          );
-          examples = pkgs.lib.flip pkgs.lib.mapAttrs (builtins.readDir ./examples) (
-            name: _:
-              mkTest {
-                name = "test ${name}";
-                script = ''
-                  cp -r ${./.}/examples/${name} ./${name}
-                  chmod -R u+w ./${name}
-                  cd ./${name}
-                  nix flake lock --override-input nickel-nix path:${./.}
-                  nix run .#regenerate-lockfile --accept-flake-config
-                  nix build --accept-flake-config
-                '';
-              }
-          );
+        apps.run-test =
+          let testScript = pkgs.writeShellApplication {
+            name = "test-templates";
+            runtimeInputs = [ inputs.nickel.packages."${system}".nickel-lang-cli ];
+            text = ''
+              set -euo pipefail
+              cd ${./.}
+              exec ${pkgs.runtimeShell} ./run-test.sh "$@"
+            '';
```

```suggestion
            text = builtins.readFile ./run-test.sh;
```

Since shell script requires this `cd` line anyway, it should happen in the script. This would also run additional checks against the script itself (like ShellCheck) so that we can be more confident in it.

---

I couldn't find a nice way of `cd`-ing directly from the script (as we can't really know the path of the flakeroot from it). But I've just dropped it as it's fair to just assume that we're already in the right directory